lib/maplibre-compose/src/maplibreNativeMain/kotlin/dev/sargunv/maplibrecompose/compose/offline/DownloadProgress.kt

```diff
@@ -0,0 +1,19 @@
+package dev.sargunv.maplibrecompose.compose.offline
+
+public sealed interface DownloadProgress {
+  public data object Unknown : DownloadProgress
+
+  public data class Normal(
+    val completedResourceCount: Long,
+    val completedResourceBytes: Long,
+    val completedTileCount: Long,
+    val completedTileBytes: Long,
+    val downloadState: DownloadState,
+    val isRequiredResourceCountPrecise: Boolean,
+    val requiredResourceCount: Long,
+  ) : DownloadProgress
+
+  public data class Error(val reason: String, val message: String) : DownloadProgress
+
+  public data class TileLimitExceeded(val limit: Long) : DownloadProgress
```

Comment:
Wouldn't this be an error? Where's the difference between an Error (with reason="Tile limit exceeded") and this one?

Reply:
I think the download can proceed once other regions are deleted

The MLN SDK differentiates between errors and this state